Sources/Build/BuildPlan.swift

```diff
@@ -139,19 +139,22 @@ public enum TargetBuildDescription {
             return target.libraryBinaryPaths
         }
     }
+
+    var resourceBundleInfoPlistPath: AbsolutePath? {
+        switch self {
+        case .swift(let target):
+            return target.resourceBundleInfoPlistPath
+        case .clang(let target):
+            return target.resourceBundleInfoPlistPath
+        }
+    }
 }
 
-/// Target description for a Clang target i.e. C language family target.
-public final class ClangTargetBuildDescription {
+public class TargetBuildDescriptionImpl {
```

Comment:
Is it possible to use some sort of helper instead of a subclassing? swiftc and clang have very different compilation model and I am afraid it's a bit risky for the future to model this using subclassing.